File: 品味Java的21種設計模式/dp_cpp/prototype/example8/ConcretePrototype1.cpp
#include "ConcretePrototype1.h"

namespace cn
{
	namespace javass
	{
		namespace dp
		{
			namespace prototype
			{
				namespace example8
				{

					std::string ConcretePrototype1::getName()
					{
						return name;
					}

					void ConcretePrototype1::setName(std::string name)
					{
						this->name = name;
					}

					Prototype *ConcretePrototype1::clone()
					{
						ConcretePrototype1 *prototype = new ConcretePrototype1();
						prototype->setName(this->name);
						return prototype;
					}

					std::string ConcretePrototype1::ToString()
					{
						return "Now in Prototype1，name="+name;
					}
				}
			}
		}
	}
}

File: 品味Java的21種設計模式/dp_java/prototype/example8/ConcretePrototype1.java
package cn.javass.dp.prototype.example8;
/**
 * 克隆的具体实现对象
 */
public class ConcretePrototype1 implements Prototype {
    private String name;
    public String getName() {
        return name;
    }
    public void setName(String name) {
        this.name = name;
    }

    public Prototype clone() {
        ConcretePrototype1 prototype = new ConcretePrototype1();
        prototype.setName(this.name);
        return prototype;
    }
    public String toString() {
        return "Now in Prototype1，name=" + name;
    }
}

